server.R

```diff
@@ -122,21 +127,23 @@ server <- function( input, output, session ) {
   # query the ACS API
   # based on the ACS Table and store the results
   user.data <- reactive({
+    
+    table.selected <- if_else(input$selectTableSlim == "Other", input$selectTable, input$selectTableSlim)
+    
     # require that the three inputs needed to fetch ACS data are not NULL
     # note: used to hide initial error message when data is loading
-    validate( need( expr = variableList$stubLong == input$variable &
-                        variableList$tableID == tableList$tableID[tableList$stub == input$select.table]
+    validate( need( expr = variables$variableName == input$variable &
+                      variables$tableStub == table.selected
                       , message = "Loading. If no data loads, make sure you have selected a table and variable" ))
 
     
     #download data
-    table.selected <- tableList$tableID[tableList$stub == input$select.table]
+    table.selectedID <- unique(variables$tableID[variables$tableStub == table.selected])
```

Comment:
`table.selectedID` should be its own `reactive()` expression.  It is used throughout the server so store it once and simply call table.select() throughout to avoid re-filtering `variables` and extracting its unique `tableID` value.

Reply:
✓